js_modules/dagster-ui/packages/ui-core/src/assets/Stale.tsx

```diff
@@ -127,50 +175,158 @@ function groupedCauses(
+}) => {
+  const grouped = groupedCauses(assetKey, include, liveData);
+  const totalCauses = liveData?.staleCauses?.length;
+
+  if (!totalCauses) {
+    // Should never happen since the parent of this component should check that the asset is stale before rendering the popover
+    return <div />;
+  }
+  return (
+    <Box flex={{direction: 'column'}}>
+      <Box padding={{horizontal: 12, vertical: 8}} border="bottom">
+        <Subtitle2>
+          {numberFormatter.format(totalCauses)} {ifPlural(totalCauses, 'change', 'changes')} since
+          last materialization
+        </Subtitle2>
+      </Box>
+      {Object.entries(grouped).map(([label, causes], idx) => {
+        const isSelf = isEqual(assetKey.path, causes[0]!.key.path);
+        return (
+          <Box key={label}>
+            <Box
+              padding={{horizontal: 12, vertical: 8}}
+              border={idx === 0 ? 'bottom' : 'top-and-bottom'}
+            >
+              <CaptionSubtitle>
+                {getCollapsedHeaderLabel(isSelf, causes[0]!.category, causes.length)}
+              </CaptionSubtitle>
+            </Box>
+            {causes.map((cause, idx) => (
+              <StaleReason
+                key={idx}
+                reason={cause.reason}
+                dependency={cause.dependency}
+                assetKey={assetKey}
+              />
+            ))}
+          </Box>
+        );
+      })}
     </Box>
-  </Box>
-);
+  );
+};
 
 const StaleReason = ({
   reason,
   dependency,
 }: {
+  assetKey: AssetKeyInput;
   reason: string;
   dependency: AssetNodeKeyFragment | null;
 }) => {
-  if (!dependency) {
-    return <Caption>{` ${reason}`}</Caption>;
-  }
+  const content = useMemo(() => {
+    if (!dependency) {
+      return <Caption>{` ${reason}`}</Caption>;
+    }
+
+    const dependencyName = displayNameForAssetKey(dependency);
+    const dependencyPythonName = dependencyName.replace(/ /g, '');
+    if (reason.endsWith(`${dependencyPythonName}`)) {
+      const reasonUpToDep = reason.slice(0, -dependencyPythonName.length);
+      return (
+        <>
+          <Caption>{reasonUpToDep}</Caption>
+          <Link to={assetDetailsPathForKey(dependency)}>
+            <Tag icon="asset">{dependencyName}</Tag>
+          </Link>
+        </>
+      );
+    }
 
-  const dependencyName = displayNameForAssetKey(dependency);
-  const dependencyPythonName = dependencyName.replace(/ /g, '');
-  if (reason.endsWith(`${dependencyPythonName}`)) {
-    const reasonUpToDep = reason.slice(0, -dependencyPythonName.length);
     return (
-      <Caption>
-        {` ${reasonUpToDep}`}
-        <Link to={assetDetailsPathForKey(dependency)}>{dependencyName}</Link>
-      </Caption>
+      <>
+        <Link to={assetDetailsPathForKey(dependency)}>
+          <Tag icon="asset">{dependencyName}</Tag>
+        </Link>
+        <Caption>{` ${reason} `}</Caption>
+      </>
     );
-  }
+  }, [dependency, reason]);
+  return (
```

Comment:
Since basically the entire thing here is memoized we could wrap the whole component in `React.memo(()=>` and avoid having to specify dependencies explicitly